File: app/src/main/java/com/github/kongpf8848/animation/MainHandler.java
package com.github.kongpf8848.animation;

import android.os.Handler;
import android.os.Looper;

public class MainHandler extends Handler {

    private static volatile MainHandler instance;

    public static MainHandler getInstance(){
        if(instance==null){
            synchronized (MainHandler.class) {
                if (instance == null) {
                    instance= new MainHandler();
                }
            }
        }
        return instance;
    }

    private MainHandler() {
        super(Looper.getMainLooper());
    }

    public void runOnUiThread(Runnable runnable) {
        if (Looper.getMainLooper()== Looper.myLooper()) {
            runnable.run();
        } else {
           post(runnable);
        }
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/utils/MainHandler.kt
package com.github.kongpf8848.animation.utils

import android.os.Handler
import android.os.Looper

class MainHandler : Handler(Looper.getMainLooper()) {
    private object MainHandler {
        val holder = MainHandler()
    }

    fun runOnUiThread(runnable: Runnable) {
        if (Looper.getMainLooper() == Looper.myLooper()) {
            runnable.run()
        } else {
            post(runnable)
        }
    }

    companion object {
        fun getInstance()=MainHandler.holder
    }
}